hypothesis-python/tests/cover/test_example.py

```diff
@@ -66,3 +73,23 @@ def test_example_inside_find():
 @fails_with(HypothesisException)
 def test_example_inside_strategy():
     st.booleans().map(lambda x: st.integers().example()).example()
+
+
+def test_non_interactive_example_emits_warning():
+    # We have this warning disabled for most of our tests, because self-testing
+    # Hypothesis means `.example()` can be a good idea when it never is for users.
+    with warnings.catch_warnings():
+        warnings.simplefilter("always")
+        with pytest.warns(NonInteractiveExampleWarning):
+            st.text().example()
+
+
+def test_interactive_example_does_not_emit_warning():
+    try:
+        child = pexpect.spawn("%s -Werror" % (sys.executable,))
+    except AttributeError:
+        pytest.skip("pexpect.spawn not supported on Windows")
```

Comment:
I slightly prefer putting the `pytest.skip` at the top level, and checking the `WINDOWS` constant from hypothesis.internal.compat.